sonic_installer/main.py

```diff
@@ -872,17 +873,17 @@ def upgrade_docker(container_name, url, cleanup_image, skip_check, tag, warm):
     # this is image_id for image with "latest" tag
     image_id_latest = get_container_image_id(image_latest)
 
-    for id in image_id_all:
-        if id != image_id_latest:
-            # Unless requested, the previoud docker image will be preserved
-            if not cleanup_image and id == image_id_previous:
-                continue
-            run_command("docker rmi -f %s" % id)
+    if cleanup_image:
+        # Unless requested, the previoud docker image will be preserved
+        for id in image_id_all:
+            if id != image_id_latest and id == image_id_previous:
```

Comment:
>id != image_id_latest and id == image_id_previous [](http://example.com/codeflow?start=15&length=49)

This check is weird, seems like `image_id_latest != image_id_previous`, and not related to the loop var. Could you double check?

Reply:
will fix